bindings/python/dlite-collection-python.i

```diff
@@ -11,24 +11,32 @@
             v = self.next()
             if v and v.is_meta:
                 v.__class__ = Metadata
-        elif self.rettype in 'RTspo':  # relation
+        elif self.rettype in 'RTtspod':  # relation
             v = self.next_relation()
         else:
-            raise ValueError('`rettype` must one of "IRTspo"')
+            raise ValueError('`rettype` must one of "IRTtspod"')
 
         if v is None:
             raise StopIteration()
 
-        if self.rettype == 'T':  # return (s,p,o) tuple
+        if self.rettype == 'I':  # return instance
+            return v
+        if self.rettype == 'R':  # return relation object
+            return v.copy()
+        if self.rettype == 'T':  # return (s,p,o,d) tuple
+            return (v.s, v.p, v.o, v.d)
+        if self.rettype == 't':  # return (s,p,o) tuple
```

Comment:
When introducing a new type of variable, isn't it better that the 'old' one stays the same as before?

Reply:
I principle I fully agree. 

But it wasn't long ago that the rettype variable was introduced. It is also quite hidden and was not documented before this PR, so I don't expect anyone using it or even recognised it. Therefore I it is ok to break backward compatibility to the benefit of more logical naming in the future (I associate capital with more).

Another more important question is whether 'T' (s,p,o,d) should be the default for get_relations() rather than 't' (s,p,o). Here I chose 't' for backward compatibility reasons.  But it makes it easy to miss the datatype. I even did it myself when updating tripper. 

What do you think. Is it worth to break compatibility for a probably more safe interface in the future? Maybe a better solution would be to add a new method in the Python interface that also returns the datatype. What would be a good name for such a method?

Reply:
I think T should be default. 
But not sure if we should go for breaking change suddenyl from one release to the next or if we should add warning in some way,

Reply:
Agree. Lets do that in a separate PR and check with the team before doing it. I will add an issue.